papers/hendrik_heuer/heuer.rst

```diff
@@ -0,0 +1,223 @@
+
+Implementation
+--------------
+
+The text comparison tool implements a workflow that consists of a Python tool for the back-end and a Javascript tool for the front-end. With the Python tool, a text is converted into a collection of two-dimensional word vectors. These are visualized using the Javascript front-end. With the Javascript front-end, the user can explore the word map and zoom in and out to investigate both the local and the global structure of the text sources. The Javascript front-end can be published online.
+
+The workflow of the tool includes the following four steps: 
+
+Pre-processing
+~~~~~~~~~~~~~~
+
+In the pre-processing step, all sentences are tokenized to extract single words. The tokenization is done using the Penn Treebank Tokenizer implemented in the Natural Language Processing Toolkit (NLTK) for Python [Bir09]. Alternatively, this could also be achieved with a regular expression.
+
+Using a hash map, all words are counted. Only unique words, i.e. the keys of the hash map, are taken into account for the dimensionality reduction. The 3000 most frequent English words according to a frequency list collected from Wikipedia are ignored to reduce the amount of data.
+
+Word representations
+~~~~~~~~~~~~~~~~~~~~
+
+For all unique non-frequent words, the word representation vectors are collected from the word2vec model from the gensim Python library [Řeh10]. Each word is represented by an N-dimensional vector (N=300). 
+
+.. code-block:: python
+
+   from gensim.models import Word2Vec
+
+   model = Word2Vec.load_word2vec_format( \
+    word_vectors_filename, binary=True )
+
+   for word in words:
+     if word in model:
+       print model[ word ]
+
+
+Dimensionality Reduction
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The resulting 300-dimensional word2vec vectors are projected down to 2D using the t-SNE Python implementation in scikit-learn [Ped11].
+
+In the dimensionality reduction step, the 300-dimensional word vectors are projected down to a two-dimensional space so that they can be easily visualized in a 2D coordinate system (see Fig. 2).
+
+.. figure:: tsne_dimensionality_reduction.png
+
+   In the dimensionality reduction step, the word vectors are projected down to 2D :label:`egfig`
+
+For the implementation, the t-SNE implementation in scikit-learn is used:
+
+.. code-block:: python
+
+   from sklearn.manifold import TSNE
+
+   tsne = TSNE(n_components=2)
+   tsne.fit_transform( word_vectors )
+
+Visualization
+~~~~~~~~~~~~~
+
+After the dimensionality reduction, the vectors are exported to a JSON file. The vectors are visualized using the D3.js JavaScript data visualization library [Bos12]. Using D3.js, an interactive map was developed. With this map, the user can move around and zoom in and out.
```

Comment:
Some more discussion of the visualization / necessity of multi-resolution stuff (zooming, moving around) would be useful - otherwise I would collapse this section into the above, since it feels pretty light. Maybe an example of "global scale" vs "local scale" features? E.g. exploring dissimilar words vs similar words